app/models/spree/image.rb

```diff
@@ -51,26 +47,23 @@ def no_attachment_errors
       false
     end
 
-    # Spree stores attachent definitions in JSON. This converts the style name and format to
-    # strings. However, when paperclip encounters these, it doesn't recognise the format.
-    # Here we solve that problem by converting format and style name to symbols.
-    # See also: ImageSettingsController decorator.
-    #
-    # eg. {'mini' => ['48x48>', 'png']} is converted to {mini: ['48x48>', :png]}
-    def self.format_styles(styles)
-      styles_a = styles.map do |name, style|
-        style[1] = style[1].to_sym if style.is_a? Array
-        [name.to_sym, style]
-      end
-
-      Hash[styles_a]
+    def self.set_attachment_attributes(attribute_name, attribute_value)
+      Spree::Image.attachment_definitions[:attachment][attribute_name] = attribute_value
     end
 
-    def self.reformat_styles
-      Spree::Image.attachment_definitions[:attachment][:styles] =
-        format_styles(Spree::Image.attachment_definitions[:attachment][:styles])
+    def self.set_s3_attachment_definitions
+      if Spree::Config[:use_s3]
+        s3_creds = { access_key_id: Spree::Config[:s3_access_key],
+                     secret_access_key: Spree::Config[:s3_secret],
+                     bucket: Spree::Config[:s3_bucket] }
+        Spree::Image.attachment_definitions[:attachment][:storage] = :s3
+        Spree::Image.attachment_definitions[:attachment][:s3_credentials] = s3_creds
+        Spree::Image.attachment_definitions[:attachment][:s3_headers] =
+          ActiveSupport::JSON.decode(Spree::Config[:s3_headers])
+        Spree::Image.attachment_definitions[:attachment][:bucket] = Spree::Config[:s3_bucket]
+      else
+        Spree::Image.attachment_definitions[:attachment].delete :storage
+      end
```

Comment:
We should replace all these `Spree::Image` with `self`

Reply:
nice :+1: actually it can be removed because then self becomes redundant (rubocop helps remembering that).